apps/astarte_pairing/lib/astarte_pairing/fdo/ownership_voucher/rendezvous_info.ex

```diff
@@ -0,0 +1,70 @@
+defmodule Astarte.Pairing.FDO.OwnershipVoucher.RendezvousInfo do
+  @moduledoc """
+  Implementation of FDO RendezvousInfo type.
+  """
+
+  alias Astarte.Pairing.FDO.OwnershipVoucher.RendezvousInfo.RendezvousDirective
+  alias Astarte.Pairing.FDO.OwnershipVoucher.RendezvousInfo.RendezvousInstr
+  alias Astarte.Pairing.FDO.OwnershipVoucher.RendezvousInfo.RVVariable
+  require Logger
+
+  @type t :: [RendezvousDirective.t()]
+
+  @doc """
+  Decode CBOR-decoded RendezvousInfo into structs.
+  """
+  @spec decode(list(list())) :: {:ok, [RendezvousDirective.t()]} | {:error, any()}
+  def decode(directives) when is_list(directives) do
+    directives
+    |> Enum.reduce_while([], fn instrs, acc ->
+      case decode_directive(instrs) do
+        {:ok, directive} -> {:cont, [directive | acc]}
+        {:error, reason} -> {:halt, {:error, reason}}
+      end
+    end)
+    |> case do
+      {:error, reason} -> {:error, reason}
+      list -> {:ok, Enum.reverse(list)}
+    end
+  end
+
+  def decode(_), do: {:error, :invalid_format}
+
+  defp decode_directive(instrs) when is_list(instrs) do
+    instrs
+    |> Enum.reduce_while([], fn
+      [u8, val], acc ->
+        case RVVariable.rv_from_u8(u8) do
+          {:ok, var} ->
+            instr = %RendezvousInstr{rv_variable: var, rv_value: val}
+            {:cont, [instr | acc]}
+
+          :error ->
+            {:halt, {:error, {:invalid_rv_variable, u8}}}
+        end
+
+      _, _acc ->
+        {:halt, {:error, :invalid_instruction_format}}
+    end)
+    |> case do
+      {:error, reason} -> {:error, reason}
+      instrs_list -> {:ok, %RendezvousDirective{instructions: Enum.reverse(instrs_list)}}
+    end
+  end
+
+  @doc "Encode RendezvousInfo into CBOR-compatible terms."
+  def encode(directives) when is_list(directives) do
+    Enum.map(directives, &encode_directive/1)
+  end
+
+  defp encode_directive(%RendezvousDirective{instructions: instrs}) do
+    Enum.map(instrs, &encode_instruction/1)
+  end
+
+  defp encode_instruction(%RendezvousInstr{rv_variable: var, rv_value: val}) do
+    [RVVariable.rv_to_u8(var), encode_rv_value(val)]
+  end
+
+  defp encode_rv_value(%CBOR.Tag{} = t), do: t
```

Comment:
I suggest adding the check that it is in fact a binary, tags can have many forms


```suggestion
  defp encode_rv_value(%CBOR.Tag{tag: :bytes} = t), do: t
```